ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -56,20 +63,45 @@
   Log(String dataDir, long totalCapacityInBytes, long segmentCapacityInBytes, StoreMetrics metrics) throws IOException {
     this.dataDir = dataDir;
     this.capacityInBytes = totalCapacityInBytes;
+    this.isLogSegmented = totalCapacityInBytes > segmentCapacityInBytes;
     this.metrics = metrics;
+    this.segmentNameAndFileNameIterator = Collections.EMPTY_LIST.iterator();
 
     File dir = new File(dataDir);
     File[] segmentFiles = dir.listFiles(LogSegmentNameHelper.LOG_FILE_FILTER);
     if (segmentFiles == null) {
       throw new IOException("Could not read from directory: " + dataDir);
-    } else if (segmentFiles.length == 0) {
-      // first startup
-      checkArgsAndAllocateFirstSegment(totalCapacityInBytes, segmentCapacityInBytes);
     } else {
-      // subsequent startup
-      loadSegments(segmentFiles, totalCapacityInBytes);
+      initialize(getSegmentsToLoad(segmentFiles), segmentCapacityInBytes);
     }
-    activeSegment = segmentsByName.lastEntry().getValue();
+  }
+
+  /**
+   * Create a Log instance
+   * @param dataDir the directory where the segments of the log need to be loaded from.
+   * @param totalCapacityInBytes the total capacity of this log.
+   * @param segmentCapacityInBytes the capacity of a single segment in the log.
+   * @param metrics the {@link StoreMetrics} instance to use.
+   * @param isLogSegmented {@code true} if this log is segmented or needs to be segmented.
+   * @param segmentsToLoad the list of pre-created {@link LogSegment} instances to load.
+   * @param segmentNameAndFileNameIterator an {@link Iterator} that provides the name and filename for newly allocated
+   *                                       log segments. Once the iterator ends, the active segment name is used to
+   *                                       generate the names of the subsequent segments.
+   * @throws IOException if there is any I/O error loading the segment files.
+   * @throws IllegalArgumentException if {@code totalCapacityInBytes} or {@code segmentCapacityInBytes} <= 0 or if
+   * {@code totalCapacityInBytes} > {@code segmentCapacityInBytes} and {@code totalCapacityInBytes} is not a perfect
+   * multiple of {@code segmentCapacityInBytes}.
+   */
+  Log(String dataDir, long totalCapacityInBytes, long segmentCapacityInBytes, StoreMetrics metrics,
```

Comment:
synced up - the reason is that this be used in the data mover of compaction that needs to provide custom log segment names and filenames.